packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st

```diff
@@ -3,8 +3,8 @@ testErrorHandling
 	|dummy|
 	client
 		at: 'thrower'
-		ifPresent: []
-		ifAbsentPut: ThrowerOfErrors new.
+		ifPresent: [self fail]
+		ifAbsentPut: LQThrowerOfErrors new.
 	dummy := client at: 'thrower' ifAbsent: [ self fail ].
 	[dummy escalate] on: RemoteError do: [^ self "everything is good" ].
```

Comment:
[sand] Theres also self should: aBlock raise: anException , which would be cleaner here IMO. 
[boulder] This test passes if no error is thrown, as I understand it this is not the intended behaviour. Could be fixed by using the suggestion above

Reply:
[sand]: You would replace line 9 with your suggestion, do I understand correctly? That way, you would circumvent `on: do:` by using `should: raise:` which would indeed be cleaner.
[boulder]: The test should not pass if no error is thrown, did you test this? dummy escalate throws a RemoteError, which leads us going into the `do:`-Block, which is what we wanted. If we pass the `do:` we `self fail` in the next line.

Reply:
Adding to the boulder: I think I see what you mean now. `dummy escalate` indeed does not throw a `RemoteError`, but a general Error - that is a mistake on my part. I have changed this and also `should: raise:`

Reply:
Ahh sorry, didn't see the self fail in the last line, so this is purely a cosmetic suggestion now: replace 
```smalltalk
[dummy escalate] on: RemoteError do: [^ self "everything is good" ].
	self fail
```
with
```smalltalk
	self
		should: [dummy escalate]
		raise: RemoteError
```
I tested it, it runs through.
Since the boulder is solved, I approved the PR

Reply:
> Adding to the boulder: I think I see what you mean now. `dummy escalate` indeed does not throw a `RemoteError`, but a general Error - that is a mistake on my part. I have changed this and also `should: raise:`

That's not what I meant, but good think you found another error this way ^^

Reply:
It seems we commented at the same time ^^

Reply:
> > Adding to the boulder: I think I see what you mean now. `dummy escalate` indeed does not throw a `RemoteError`, but a general Error - that is a mistake on my part. I have changed this and also `should: raise:`
> 
> That's not what I meant, but good think you found another error this way ^^

What happened there is that we had `[dummy escalate]` in []-brackets, which evaluated the statement in the block. It seems that that evaluation gets interrupted by the error and just returns (I tested this in a Workspace, `[dummy escalate]` doesn't throw an error...), without passing on the error. I don't however know why the test didn't fail afterwards, when we got to `self fail`

Reply:
> What happened there is that we had [dummy escalate] in []-brackets, which evaluated the statement in the block. It seems that that evaluation gets interrupted by the error and just returns (I tested this in a Workspace, [dummy escalate] doesn't throw an error...), without passing on the error. I don't however know why the test didn't fail afterwards, when we got to self fail

Correction: The code worked as it should have. The difference is that when the client is a `RemoteRepoClient` `dummy escalate` raises a RemoteError, whilst a `LocalObjectRepo` raises a normal Error. My tests seem to have used different Repos, which is why the test didn't work as it would normally. So the "error" that I found wasn't an error, but just an oversight when extending the errorHandling-Test to the LocalObjectRepo